rusk/src/lib/node/rusk.rs

```diff
@@ -465,7 +465,32 @@ impl Rusk {
         self.query(STAKE_CONTRACT, "get_stake", pk)
     }
 
-    pub(crate) fn session(
+    /// Opens a session for a new block proposal/verification.
+    ///
+    /// Before returning the session, "before_state_transition" of Stake
+    /// Contract is called
+    pub(crate) fn new_block_session(
+        &self,
+        block_height: u64,
+        commit: Option<[u8; 32]>,
+    ) -> Result<Session> {
+        let mut session = self._session(block_height, commit)?;
+        let _: CallReceipt<()> = session
+            .call(STAKE_CONTRACT, "before_state_transition", &(), u64::MAX)
+            .expect("before_state_transition to success");
+        Ok(session)
+    }
+
+    /// Opens a session for query, setting a block height of zero since this
+    /// doesn't affect the result.
+    pub(crate) fn query_session(
+        &self,
+        commit: Option<[u8; 32]>,
+    ) -> Result<Session> {
+        self._session(0, commit)
+    }
+
+    pub(crate) fn _session(
```

Comment:
Is the trailing `_` necessary?

Reply:
It's a common way to let the dev know to not directly call the function (neither locally)

Reply:
Btw, I removed the `pub(crate)` accessor and added a comment to warn users to not call it directly